### PR TITLE
Issue 15

### PR DESCRIPTION
**Pull request by stefanhahmann**
This PR adds a new feature on Spot Level - the SpotEllipsoidAspectRatioFeature that computes aspect ratio of the ellipsoid axes.

### CODE REVIEW

File: src/test/java/org/mastodon/mamut/feature/spot/ellipsoid/AbstractEllipsoidFeatureTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import org.mastodon.feature.FeatureProjection;
+import org.mastodon.feature.FeatureProjectionKey;
+import org.mastodon.feature.FeatureProjectionSpec;
+import org.mastodon.mamut.model.Model;
+import org.mastodon.mamut.model.ModelGraph;
+import org.mastodon.mamut.model.Spot;
+
+import java.io.IOException;
+
+public abstract class AbstractEllipsoidFeatureTest
```
Comment (maarzt) [inline src/test/java/org/mastodon/mamut/feature/spot/ellipsoid/AbstractEllipsoidFeatureTest.java:13]: This does not need to be an abstract class. I think you a better off with a normal "SingleEllipsoidExampleGraph" class.

**Review by maarzt — APPROVED** [review]

File: src/main/java/org/mastodon/mamut/feature/branch/leaves/BranchNLeavesFeatureComputer.java [hunk 2]
```diff
@@ -16,7 +17,7 @@
  * Computes {@link BranchNLeavesFeature}
  */
 @Plugin( type = MamutFeatureComputer.class )
-public class BranchNLeavesFeatureComputer implements MamutFeatureComputer
+public class BranchNLeavesFeatureComputer extends CancelableImpl implements MamutFeatureComputer
```
[the remark this answers is not in the document]
Comment (stefanhahmann) [reply src/main/java/org/mastodon/mamut/feature/branch/leaves/BranchNLeavesFeatureComputer.java:20]: Well spotted. I added checking for isCanceled to the FeatureComputation. To do so, the method LineageTreeUtils.callDepthFirst() was extended with BooleanSupplier that forwards the cancel state to the iteration.

File: src/main/java/org/mastodon/mamut/feature/branch/sinuosity/BranchSinuosityFeatureComputer.java [hunk 3]
```diff
@@ -17,7 +18,7 @@
  * Computes {@link BranchSinuosityFeature}
  */
 @Plugin( type = MamutFeatureComputer.class )
-public class BranchSinuosityFeatureComputer implements MamutFeatureComputer
+public class BranchSinuosityFeatureComputer extends CancelableImpl implements MamutFeatureComputer
```
[the remark this answers is not in the document]
Comment (stefanhahmann) [reply src/main/java/org/mastodon/mamut/feature/branch/sinuosity/BranchSinuosityFeatureComputer.java:21]: Thanks for the advice. I added isCanceled() in the computation for loop.

File: src/main/java/org/mastodon/mamut/feature/branch/successors/BranchNSuccessorsFeatureComputer.java [hunk 4]
```diff
@@ -16,7 +17,7 @@
  * Computes {@link BranchNSuccessorsFeature}
  */
 @Plugin( type = MamutFeatureComputer.class )
-public class BranchNSuccessorsFeatureComputer implements MamutFeatureComputer
+public class BranchNSuccessorsFeatureComputer extends CancelableImpl implements MamutFeatureComputer
```
[the remark this answers is not in the document]
Comment (stefanhahmann) [reply src/main/java/org/mastodon/mamut/feature/branch/successors/BranchNSuccessorsFeatureComputer.java:20]: Well spotted. I added checking for isCanceled to the FeatureComputation. To do so, the method LineageTreeUtils.callDepthFirst() was extended with BooleanSupplier that forwards the cancel state to the iteration.

File: src/test/java/org/mastodon/mamut/feature/spot/ellipsoid/AbstractEllipsoidFeatureTest.java (hunk 1 above)
Comment (stefanhahmann) [reply src/test/java/org/mastodon/mamut/feature/spot/ellipsoid/AbstractEllipsoidFeatureTest.java:13]: I would like to keep this class for more EllipsoidFeatureTests to come. I thought about an extra SingleEllipsoidExampleGraph, but it does not seem to simplify things at the moment

**Pull request merged**